v5/core/base_service.go

```diff
@@ -273,41 +300,9 @@ func (service *BaseService) IsSSLDisabled() bool {
 	return false
 }
 
-// getClientTransportForSSL() will return the http.Transport instance
-// that needs to be modified to disable SSL.  This is a bit tricky
-// because we have to account for the retries-enabled scenario.
-func getClientTransportForSSL(client *http.Client) *http.Transport {
-	// "client" will be passed in as the client instance that is hanging off
-	// the BaseService.   This could be either a "normal" http.Client instance
-	// to be used when retries are not enabled, or it could be a "shim" http.Client
-	// instance that's used when retries are enabled.
-	// We determine which it is by checking the client's Transport field.
-	// If it is a "shim" client instance then the Transport field will be an
-	// instance of the retryablehttp.RoundTripper struct, otherwise the Transport
-	// field will be an instance of http.Transport.
-
-	// if "client" is a shim http.Client instance to support retries, then just
-	// change "client" to point to the real http.Client instance that is embedded inside
-	// the retryablehttp.Client instance, since this is the one used to invoke
-	// individual requests when retries are enabled.
-	if tr, ok := client.Transport.(*retryablehttp.RoundTripper); tr != nil && ok {
-		client = tr.Client.HTTPClient
-	}
-
-	// Next,
-	if client != nil {
-		if tr, ok := client.Transport.(*http.Transport); tr != nil && ok {
-			return tr
-		}
-	}
-
-	return nil
-}
-
 // setMinimumTLSVersion sets the minimum TLS version required by the client to TLS v1.2
 func setMinimumTLSVersion(client *http.Client) {
-	tr := getClientTransportForSSL(client)
-	if tr != nil {
+	if tr, ok := client.Transport.(*http.Transport); tr != nil && ok {
```

Comment:
we will now only call this method on an actual (non-retryable) http.Client instance